src/extract/mod.rs

```diff
@@ -12,6 +12,9 @@ pub mod greedy_dag;
 #[cfg(feature = "ilp-cbc")]
 pub mod ilp_cbc;
 
+// Allowance for floating point values to be considered equal
+pub const EPSILON_ALLOWANCE: f64 = 0.00001;
```

Comment:
I wonder if we should be using bignums for cost... idk, it might slow things down.
However it's very easy to just drop nodes when counting the cost in floats.

Anyways, this is a fine solution for now

Reply:
Great comment. Yes, it's not ideal. In our test egraphs we've got lots of small costs, so having a little allowance is OK. But if the costs were large, we'd need to have a big allowance (e.g. 2^50) to handle the imprecise nature of fp64.

I'll checkout some better solution.

Reply:
Thanks! doesn't have to go in this PR of course
I think costs get big for tree size

Reply:
Ok. Cool. I'll save a better solution for another PR.